common/src/main/scala/explore/components/graphql/SubscriptionRender.scala

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.components.graphql
+
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.vdom.html_<^._
+import react.common._
+import react.semanticui.elements.icon.Icon
+import react.semanticui.sizes._
+import clue.GraphQLStreamingClient
+import cats.implicits._
+import cats.effect.IO
+import crystal._
+import crystal.react.StreamRenderer
+import crystal.react.io.implicits._
+import diode.data._
+import diode.react.ReactPot._
+import cats.effect.ContextShift
+
+final case class SubscriptionRender[D, A](
+  subscribe:      IO[GraphQLStreamingClient[IO]#Subscription[D]],
+  streamModifier: fs2.Stream[IO, D] => fs2.Stream[IO, A] = identity[fs2.Stream[IO, D]] _
+)(
+  val valueRender: A => VdomNode,
+  val onNewData:   IO[Unit] = IO.unit
+)(
+  implicit val cs: ContextShift[IO]
+) extends ReactProps {
+  override def render: VdomElement =
+    SubscriptionRender.component(this.asInstanceOf[SubscriptionRender[Any, Any]])
```

Comment:
Why's the casting needed?

Reply:
It is needed because the component has type parameters, and therefore has to be defined in a `def`. But we don't want it to be created in each rerender, so we store in a `val` an instance parametrized on `[Any, Any]`, which can be reused with the casting. I'm open to more elegant ways of solving this problem.